### PR TITLE
Feat/pap campaign bdv filters

### PR DESCRIPTION
**Pull request by elvino29**

### CODE REVIEW

File: src/DataFixtures/ORM/LoadPapAddressData.php [hunk 1]
```diff
@@ -126,6 +181,122 @@ public function load(ObjectManager $manager)
...
+            48.760296,
+            2.29762
+        );
+        $address->addZone(LoadGeoZoneData::getZoneReference($manager, 'zone_city_92002'));
+        $address->addVoter($this->createVoter(self::VOTER_13_UUID, 'Kevin', 'HEDIEUX', Genders::MALE, '-60 years', '92002_0002'));
+        $address->votePlace = $vpAnthonyb;
+        $this->addReference('address-92-4', $address);
+        $manager->persist($address);
+
+//        $address = $this->createAddress(
```
Comment (Remg) [inline src/DataFixtures/ORM/LoadPapAddressData.php:258]: either remove or use this code

File: src/Entity/Pap/Campaign.php [hunk 2]
```diff
@@ -195,6 +195,46 @@ class Campaign implements IndexableEntityInterface, EntityScopeVisibilityInterfa
      */
     private $votePlaces;
 
+    /**
+     * @ORM\Column(name="delta_prediction_and_result_min_2017", type="float", nullable=true)
+     */
+    private ?float $deltaPredictionAndResultMin2017;
```
Comment (Remg) [inline src/Entity/Pap/Campaign.php:201]:
```suggestion
    private ?float $deltaPredictionAndResultMin2017 = null;
```

File: src/Entity/Pap/Campaign.php [hunk 3]
```diff
@@ -195,6 +195,46 @@ class Campaign implements IndexableEntityInterface, EntityScopeVisibilityInterfa
...
+
+    /**
+     * @ORM\Column(name="delta_prediction_and_result_max_2017", type="float", nullable=true)
+     */
+    private ?float $deltaPredictionAndResultMax2017;
```
Comment (Remg) [inline src/Entity/Pap/Campaign.php:206]:
```suggestion
    private ?float $deltaPredictionAndResultMax2017 = null;
```

same for other fields

File: src/Repository/Pap/AddressRepository.php [hunk 4]
```diff
@@ -27,15 +27,25 @@ public function findNear(
...
                     AND 2 * (17 - :zoom) * FLOOR((1.0 - LN(TAN(RADIANS(:latitude)) + 1.0 / COS(RADIANS(:latitude))) / PI()) / 2.0 * (1 << :zoom) + 1)
+                And address.vote_place_id IN (
+SQL;
+        $comma = '';
+        for($i=0, $iMax = count($votePlaces); $i< $iMax; $i++){
+            $sql .= $comma. $votePlaces[$i];
+            $comma = ', ';
+        }
+        $sql .= ") ";
+        $sql .= <<<SQL
```
Comment (Remg) [inline src/Repository/Pap/AddressRepository.php]:
```suggestion
                And address.vote_place_id IN (
SQL;
        $sql .= implode(', ', $votePlaces);
        $sql .= ") ";
```

File: src/Repository/Pap/AddressRepository.php [hunk 5]
```diff
@@ -27,15 +27,25 @@ public function findNear(
...
                     AND 2 * (17 - :zoom) * FLOOR((1.0 - LN(TAN(RADIANS(:latitude)) + 1.0 / COS(RADIANS(:latitude))) / PI()) / 2.0 * (1 << :zoom) + 1)
+                And address.vote_place_id IN (
+SQL;
+        $comma = '';
+        for ($i = 0, $iMax = \count($votePlaces); $i < $iMax; ++$i) {
+            $sql .= $comma.$votePlaces[$i];
+            $comma = ', ';
+        }
+        $sql .= ') ';
+        $sql .= <<<SQL
```
Comment (Remg) [inline src/Repository/Pap/AddressRepository.php:48]:
```suggestion
                And address.vote_place_id IN (
SQL;
        $sql .= implode(', ', $votePlaces);
        $sql .= ') ';
```

File: src/Repository/Pap/CampaignRepository.php [hunk 6]
```diff
@@ -62,13 +62,60 @@ public function countActiveCampaign(): int
...
+        $queryBuilder = $this->createQueryBuilder('campaign')
+            ->select('DISTINCT votePlace.id')
+            ->innerJoin('campaign.votePlaces', 'votePlace')
+            ->where('campaign.finishAt IS NOT NULL AND campaign.finishAt > :now')
+            ->setParameter('now', new \DateTime())
+        ;
+
+        if ($activeCampaign) {
+            if ($deltaPredictionAndResultMin2017 = $activeCampaign->getDeltaPredictionAndResultMin2017()) {
+                $queryBuilder->andWhere(sprintf('votePlace.deltaPredictionAndResult2017 >= %s', $deltaPredictionAndResultMin2017));
```
Comment (Remg) [inline src/Repository/Pap/CampaignRepository.php:82]:
```suggestion
                $queryBuilder->andWhere('votePlace.deltaPredictionAndResult2017 >= :deltaPredictionAndResultMin2017');
                $queryBuilder->setParameter('deltaPredictionAndResultMin2017', $deltaPredictionAndResultMin2017);
```

same for other filters

**Review by mlabarrere — APPROVED** [review]

**Review by Remg — APPROVED** [review]

**Pull request merged**